libraries/AP_BattMonitor/AP_BattMonitor_Backend.cpp

```diff
@@ -214,4 +217,47 @@ void AP_BattMonitor_Backend::check_failsafe_types(bool &low_voltage, bool &low_c
     } else {
         low_capacity = false;
     }
+
+    // check battery cell voltages
+    if (has_cell_voltages()) {
+
+        uint16_t cell_diff_mv = 0.0f;
+        uint16_t cell_lowest_mv = _state.cell_voltages.cells[0];  // initialize with the first cell's voltage
+        uint16_t cell_highest_mv = _state.cell_voltages.cells[0]; // initialize with the first cell's voltage
```

Comment:
This could be UINT16_MAX in which case you will always fail (in the case of the Maxell driver suffering from noise on the bus). Just start this out with 0, since you already loop through the first one anyways.

Reply:
OK